src/views/pull_requests.py

```diff
@@ -0,0 +1,128 @@
+import time
+
+import streamlit as st
+from github.PullRequest import PullRequest
+
+from config import GithubConfig
+from pull_requests import fetch_pull_requests
+from views.sidebar.search import pull_request_search_inputs
+
+
+def pr_fetch_view(token: str) -> None:
+    fetch_pr_filter = pull_request_search_inputs()
+
+    check_github_action = st.sidebar.checkbox("Github Action Status", value=False)
+    if check_github_action:
+        st.sidebar.warning("Note that this option is slow due to multiple API calls.")
+
+    if st.sidebar.button("Fetch Pull Requests"):
+        fetch_status = st.info("Fetching pull requests, please wait!")
+        pull_requests = fetch_pull_requests(fetch_pr_filter, token)
+        st.session_state.pull_requests = pull_requests
+
+        fetch_status.info("All pull requests fetched!")
+
+    selection_result = {}
+
+    select_all = st.checkbox("Select/Deselect All", value=False)
+
+    with st.form("pr_selection"):
+        if st.session_state.pull_requests:
+            st.write(f"{len(st.session_state.pull_requests)} pull requests found!")
+        else:
+            st.write("Use the sidebar to fetch pull requests!")
+        for pr in st.session_state.pull_requests:
+            repo_name_link = f"[{pr.base.repo.full_name}/{pr.number}]({pr.html_url})"
+            if check_github_action:
+                mergability = f"{'✅ Mergable' if _is_ready_to_merge(pr) else '❌ Not Mergable'}"
+            else:
+                mergability = ""
+            needs_rebase = f"{' | ⚠️ Rebase required' if not pr.mergeable else ''}"
+            checked = st.checkbox(
+                f"{repo_name_link} | {pr.title} by {pr.user.login} | {mergability}{needs_rebase}",
+                value=select_all,
+            )
+
+            selection_result[pr] = checked
```

Comment:
**suggestion:** this forms looks like a prime candidate to become a helper function